File: bench/dotnet/spawn.cs
// Fan out N tasks, each returning its index; WhenAll and sum.
// Task.Run schedules work on the ThreadPool — .NET's lightweight concurrent
// unit. Checksum = N*(N-1)/2.
namespace Bench;
static class Spawn
{
    public static async Task Run(int n)
    {
        var tasks = Enumerable.Range(0, n)
            .Select(i => Task.Run(() => (long)i))
            .ToArray();
        var results = await Task.WhenAll(tasks);
        Console.WriteLine(results.Sum());
    }
}

File: bench/ruby/spawn.rb
# Fan out N threads, each returning its index; join and sum.
# Thread is Ruby's lightest concurrent unit. The GVL prevents CPU parallelism
# but threads still give correct concurrent semantics and test thread
# creation/join overhead. Checksum = N*(N-1)/2.
n = (ENV["BENCH_N"] || "20000").to_i

threads = Array.new(n) { |i| Thread.new { i } }
puts threads.sum(&:value)
